src/resources/elements/NetworkFeedback/NetworkFeedback.ts

```diff
@@ -20,15 +20,20 @@ export class NetworkFeedback {
     this.network = EthereumService.targetedNetwork;
     this.show = false;
 
-    this.isProduction = process.env.NODE_ENV !== "development" || process.env.NETWORK === "mainnet";
-    const storedNetwork = this.storageService.lsGet<AllowedNetworks>("network");
-    const isMainnet = [Networks.Mainnet, Networks.Celo, Networks.Arbitrum].includes(storedNetwork);
-    const isTestnet = [Networks.Rinkeby, Networks.Alfajores, Networks.Kovan].includes(storedNetwork);
+    this.isProductionEnv = process.env.NODE_ENV !== "development" || process.env.NETWORK === Networks.Mainnet;
+    const locallyStoredNetwork = this.storageService.lsGet<AllowedNetworks>("network");
+    const isLocallyMainnet = [Networks.Mainnet, Networks.Celo, Networks.Arbitrum].includes(locallyStoredNetwork);
+    const isLocallyTestnet = [Networks.Rinkeby, Networks.Alfajores, Networks.Kovan].includes(locallyStoredNetwork);
     if (
-      storedNetwork &&
-      (this.isProduction && !isMainnet) || (!this.isProduction && !isTestnet)
+      locallyStoredNetwork &&
+      ((this.isProductionEnv && isLocallyTestnet) || (!this.isProductionEnv && isLocallyMainnet))
     ) {
-      this.storageService.lsRemove("network");
+      /**
+      * If stored network is illegal, should be treated as if not selected yet,
+      * thus Ethereum network is the default.
+      */
+      this.network = this.isProductionEnv ? Networks.Mainnet : Networks.Rinkeby;
+      this.storageService.lsSet("network", `${this.network}`);
```

Comment:
Don't need comment, descriptive var name enough.

```suggestion
      const defaultNetwork = this.isProductionEnv ? Networks.Mainnet : Networks.Rinkeby;
      this.network = defaultNetwork
      this.storageService.lsSet("network", `${this.network}`);
```